x/gov/keeper/tally_handler.go

```diff
@@ -0,0 +1,144 @@
+package keeper
+
+import (
+	"cosmossdk.io/math"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	v1 "github.com/cosmos/cosmos-sdk/x/gov/types/v1"
+	stakingtypes "github.com/cosmos/cosmos-sdk/x/staking/types"
+)
+
+var _ v1.TallyHandler = DefaultTallyHandler{}
+
+// DefaultTallyHandler is the default tally handler for x/gov
+type DefaultTallyHandler struct {
+	keeper Keeper
+}
+
+// NewDefaultTallyHandler creates a new tally handler.
+func NewDefaultTallyHandler(k Keeper) DefaultTallyHandler {
+	return DefaultTallyHandler{
+		keeper: k,
+	}
+}
+
+// TODO: Break Tally into several smaller functions for clarity
+
+// Tally iterates over the votes and updates the tally of a proposal based on the voting power of the
+// voters
+func (dth DefaultTallyHandler) Tally(
```

Comment:
@DracoLi wasn't sure if this functionality is covered by existing unit or end-to-end tests -> started this doc https://kava-labs.atlassian.net/wiki/spaces/ENG/pages/1357512706/Critical+Patches+to+Review+Test and added it here for looking at adding test coverage in future sprint